### PR TITLE
[PATCH API-NEXT v14] IPsec IPv6 implementation

### PR DESCRIPTION
**Pull request by lumag**
This adds support for handling of IPv6 packets and IPv6 tunnels.

### CODE REVIEW

**Comment by lumag** [comment]
@muvarov it seems you have an underquoting in your scripts:
> muvarov changed the title from [PATCH API-NEXT v1] IPsec IPv6 implementation to [PATCH API-NEXT v7] IPsec IPv6 implementation 14 hours ago

**Comment by muvarov** [comment]
@lumag ah yes, https://github.com/muvarov/githubscripts/blob/master/gh-hook-mr.py  PATCH IPv6 also matches regexp :)

**Comment by lumag** [comment]
Updated PR to remove conflict.

**Comment by joseppc** [comment]
Looks like you forgot to squash the third commit

**Comment by lumag** [comment]
@joseppc yes, should be fixed now

**Comment by muvarov** [comment]
@versioning script fixed.

**Comment by lumag** [comment]
rebased to solve conflict

File: platform/linux-generic/odp_ipsec.c [hunk 1]
```diff
@@ -158,7 +158,7 @@ static inline int _odp_ipv4_csum_update(odp_packet_t pkt)
 					2, &chksum);
 }
 
-#define ipv4_hdr_len(ip) (_ODP_IPV4HDR_IHL(ip->ver_ihl) * 4)
+#define ipv4_hdr_len(ip) (_ODP_IPV4HDR_IHL((ip)->ver_ihl) * 4)
```
[the remark this answers is not in the document]
Comment (lumag) [reply platform/linux-generic/odp_ipsec.c]: @Bill-Fischofer-Linaro no, I'm going to use it as `ipv4_hdr_len(&ipv4hdr)`

File: test/validation/api/ipsec/ipsec_test_in.c [hunk 2]
```diff
@@ -947,6 +947,7 @@ static void test_in_ipv4_mcgrew_gcm_4_esp(void)
 	ipsec_sa_destroy(sa);
 }
 
+#if 0
```
[the remark this answers is not in the document]
Comment (lumag) [reply test/validation/api/ipsec/ipsec_test_in.c]: @Bill-Fischofer-Linaro It is basically exlained in commit log: `Test 12 is disabled till NoNH packets are properly supported in a defined way`. These are short (empty) packets. In the IETF draft they (incorrectly) bear the NH=ICMP. I plan to reenable this test setting NH=NoNH once RX of such packets is supported.

**Comment by lumag** [comment]
@psavol @Bill-Fischofer-Linaro @JannePeltonen  ping

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag I've already given my review. We'll close on this on Monday's ARCH call.

**Comment by Bill-Fischofer-Linaro** [comment]
@lumag A brief summary patch as to what's new in v12 would be helpful. I assume this is just minor refinements?

**Comment by lumag** [comment]
@Bill-Fischofer-Linaro yes, just added few lines to pass checksum checking requirements to packet parsing code (last commit).

**Comment by muvarov** [comment]
@lumag  Dmitry I tried to do my best on reviewing patches. But it's quite hard to review protocol code without directly compare it to standards. Code is merged  I hope we will have more eyes on it when it's merged.

**Pull request closed without merging**